### PR TITLE
Some work on making groups embedded

### PR DESCRIPTION
**Pull request by mateuszbaran**
Related to https://github.com/JuliaManifolds/Manifolds.jl/pull/249#issuecomment-725789794 . This seems so mostly work but there are some failures with `get_vector` and `get_coordinates` (it looks like there is some problem with dispatch transparency). @kellertuer , could you help with that?

### CODE REVIEW

**Comment by mateuszbaran** [comment]
Alternatively we could also leave `AbstractGroupManifold` as-is and add `AbstractEmbeddedGroupManifold` but that would mean we have two separate type hierarchies for group manifolds (since `AbstractEmbeddedGroupManifold` can't inherit both `AbstractGroupManifold` and `AbstractEmbeddedManifold`).

**Comment by kellertuer** [comment]
no I think the current approach is the good one, you already convinced me. I will take look probably in the evening.

**Comment by kellertuer** [comment]
While for Embedded Manifolds we surely do not want `get_vector!` and `get_coordinates!` to be transparent, for groups we surely want that - so I changed that.

Now only an `is_tangent_vector` test for `special_orthogonal` seems to fail and I don't see directly why.

**Comment by mateuszbaran** [comment]
I've slightly increased tolerance and that special orthogonal test passed locally for me.

File: src/manifolds/Rotations.jl [hunk 1]
```diff
@@ -79,7 +79,10 @@ function check_manifold_point(M::Rotations{N}, p; kwargs...) where {N}
         return DomainError(det(p), "The determinant of $p has to be +1 but it is $(det(p))")
     end
     if !isapprox(transpose(p) * p, one(p); kwargs...)
-        return DomainError(norm(p), "$p has to be orthogonal but it's not")
+        return DomainError(
+            norm(transpose(p) * p - one(p)),
+            "$p has to be orthogonal but it's not at kwargs $kwargs",
```
Comment (sethaxen) [inline src/manifolds/Rotations.jl:84]:
```suggestion
            "$p must be orthogonal but it's not at kwargs $kwargs",
```

File: test/groups/special_orthogonal.jl [hunk 2]
```diff
@@ -90,7 +90,7 @@ include("group_utils.jl")
             inverse_retraction_methods = inverse_retraction_methods,
             exp_log_atol_multiplier = 20,
             retraction_atol_multiplier = 12,
-            is_tangent_atol_multiplier = 1,
+            is_tangent_atol_multiplier = 1.2,
```
Comment (sethaxen) [inline test/groups/special_orthogonal.jl:93]: Any ideas why the change in tolerance is necessary? I'm wondering if a different default is being hit now than before.

Comment (mateuszbaran) [reply test/groups/special_orthogonal.jl:93]: I have no idea. It's a very small change so maybe there is no reason to worry about it?

Comment (sethaxen) [reply test/groups/special_orthogonal.jl:93]: Probably!

**Review by sethaxen — APPROVED** [review]
I think this just needs a patch version bump and then looks good to merge to me. Thanks!

**Review by kellertuer — APPROVED** [review]
If you have a moment of time to look for the one line the tests are not covering, that would be great, but in general I approve this.

**Comment by mateuszbaran** [comment]
I've looked at a few coverage failures and they look like false positives so I'll leave them.
